### PR TITLE
Resolved The id of the application is duplicates of the drafts are removed issue

### PR DESCRIPTION
**Pull request by yashodgayashan**
### Approach 

* Save the count of each type application in application metadata collection.
* Count the application metadata count and add it to the application id.

### Solution

* Application metadata count is not altered when applications are removed then newly added applications will get unique ID.

### CODE REVIEW

File: core/src/env_platform/db_handler.bal [hunk 1]
```diff
@@ -22,6 +23,8 @@ mongodb:Collection usersCollection = check mongoDatabase->getCollection("users")
 # there's an error while generating the applicationId.
 function saveApplication(TreeRemovalForm form) returns boolean|error {
 
+    boolean result = check saveApplicationMetadata(form.title);
+    log:printDebug("Save information in application metadata: " + result.toString());
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:27]:
```suggestion
    log:printDebug("Saved information in application metadata: " + result.toString());
```

File: core/src/env_platform/db_handler.bal [hunk 2]
```diff
@@ -263,3 +269,25 @@ function removeApplicationInUser(string userId, string applicationId) returns bo
         return error("Invalid User", message = "Couldn't find the user with given User ID");
     }
 }
+
+# The `saveApplicationMetadata` function will the number of applications of the given type by one.
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:273]:
```suggestion
# The `saveApplicationMetadata` function will save application metadata to the database.
```

File: core/src/env_platform/db_handler.bal [hunk 3]
```diff
@@ -263,3 +269,25 @@ function removeApplicationInUser(string userId, string applicationId) returns bo
...
+# 
+# + applicationType - Type of the application.
+# + return - This function will return either application meta data is added or
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:276]:
```suggestion
# + return - This function will return either whether the application meta data is added or 
```

**Comment by deshankoswatte** [comment]
@yashodgayashan have you tested this thoroughly?

**Comment by yashodgayashan** [comment]
> @yashodgayashan have you tested this thoroughly?

Yes. I have gone through each and every case and found the bug and then I resolved it and tested again manually.

**Comment by yashodgayashan** [comment]
@sanjulamadurapperuma @vikumkbv please review this and approve.

**Comment by yashodgayashan** [comment]
@vikumkbv Please review this and merge it.

**Pull request merged**